### PR TITLE
Partial compilation

### LINKED ISSUE #449: Lightdash works, even if there are some errors in the dbt compilation.
Basically, if I have a model in my project that's broken, but what I'm working on has nothing to do with that model, I should be able to still build stuff in Lightdash.

This one's a bit nuanced: and we'll have to work on figuring out a better line for what's a critical failure vs. a non-critical failure when dbt is compiling and if we differentiate between these.

### PR DESCRIPTION
**Pull request by owlas**
Closes #449 

https://www.loom.com/share/4616ff4de8de4c41aadbbe3420a25fe6

### CODE REVIEW

**Comment by owlas** [comment]
Failing e2e test is because of a problem that's been surfaced in the jaffle shop repo: https://github.com/lightdash/jaffle_shop

I'll update there and refresh this PR

**Comment by owlas** [comment]
@ZeRego e2e tests working now after update to jaffle shop

File: packages/backend/src/projectAdapters/dbtBaseProjectAdapter.ts [hunk 1]
```diff
@@ -64,6 +96,25 @@ export abstract class DbtBaseProjectAdapter implements ProjectAdapter {
         return this.rpcClient.runQuery(sql);
     }
 
+    static _isDbtModelMetadataValid(
```
Comment (ZeRego) [inline packages/backend/src/projectAdapters/dbtBaseProjectAdapter.ts:99]: I always expect a boolean return from functions that start with `isX`.
I would have called this `validateDbtModelMetadata`

File: packages/backend/src/projectAdapters/dbtBaseProjectAdapter.ts [hunk 2]
```diff
@@ -17,44 +23,70 @@ export abstract class DbtBaseProjectAdapter implements ProjectAdapter {
...
+                name: value.model.name,
+                errors: [
+                    { type: 'MetadataParseError', message: value.errorMessage },
+                ],
+            }));
+
+        // Be lazy and try to attach types to the remaining models without refreshing the catalog
+        const validModels = validatedModels
+            .filter((v) => v.isValid)
+            .map((v) => v.model);
```
Comment (ZeRego) [inline packages/backend/src/projectAdapters/dbtBaseProjectAdapter.ts:61]: We are looping 3 times over the same data. 
The validate function could handle this in a single loop. 

```
const [validModels, invalidModels] = DbtBaseProjectAdapter._validateDbtModels(models);
```

```
(models) => {

    return models.reduce(([valid, invalid], model) => {
         const model = ... 
         
         if (model.isValid) {
              return [[...valid, model], invalid];
         } else {
             return [valid, [...invalid, model]];
         }
    }, [[],[]]);
}
```

**Comment by owlas** [comment]
@ZeRego updated!

**Review by ZeRego — APPROVED** [review]
LGTM 🎉

**Pull request merged**